### PR TITLE
Add WS DPL DriverClient.

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
Do not merge for now.

**Review by davidrohr — COMMENTED** [review]
Do we really need copies for base64 and sha1, and isn't sha1 outdated nowadays?

**Comment by ktf** [comment]
If we want to be compliant with WebSockets,  SHA1 is what is required by the associated RFC. Given the usage is not / will not be particularly heavy, I think having our own copy (copied from some Public Domain source) is better than having to link against OpenSSL. 

My idea would be to reuse the same code also to allow the GUI / some level of HTML based inspector to work remotely via a browser, potentially via some SOCKS proxy.

File: Framework/Core/src/Base64.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_FRAMEWORK_BASE64_H_
+#define O2_FRAMEWORK_BASE64_H_
+
+int base64_encode(char* dest, int size, unsigned char* src, int slen);
```
Comment (sawenzel) [inline Framework/Core/src/Base64.h]: should this be in the top level namespace?

**Comment by davidrohr** [comment]
ok, I see. Perhaps we could create some kind of 3rd party folder to collect external code, and exclude that from clang-format and the codechecker, e.g. also the Utilities/Tools/cpulimit. Since I am not sure whether putting our copyright statement into such code is correct.

File: Framework/Core/src/Base64.h (hunk 1 above)
Comment (ktf) [reply Framework/Core/src/Base64.h]: does it matter? it's not a public API.

Comment (ktf) [reply Framework/Core/src/Base64.h]: anyways, I moved it in a namespace.

**Comment by ktf** [comment]
In this particular case, the copyright statement is correct since the code was purposely copied from [Public Domain](https://en.wikipedia.org/wiki/Public-domain_software). The `third_party` folders sounds nevertheless as a good idea.

**Pull request merged**